vpr/src/device/rr_graph_builder.h

```diff
@@ -218,6 +220,31 @@ class RRGraphBuilder {
     inline void init_fan_in() {
         node_storage_.init_fan_in();
     }
+    /** @brief Return detailed routing segment information with a given id* @note The routing segments here may not be exactly same as those defined in architecture file. They have been
+     * adapted to fit the context of routing resource graphs.
+     */
+
+    inline const t_segment_inf& rr_segments(RRSegmentId seg_id) const {
+        return rr_segments_[seg_id];
+    }
+    /** @brief  Return the switch information that is categorized in the rr_switch_inf with a given id
+     * rr_switch_inf is created to minimize memory footprint of RRGraph classs
+     * While the RRG could contain millions (even much larger) of edges, there are only
+     * a limited number of types of switches.
+     * Hence, we use a flyweight pattern to store switch-related information that differs
+     * only for types of switches (switch type, drive strength, R, C, etc.).
+     * Each edge stores the ids of the switch that implements it so this additional information
+     * can be easily looked up.
+     *
+     * @note All the switch-related information, such as R, C, should be placed in rr_switch_inf
+     * but NOT directly in the edge-related data of RRGraph.
+     * If you wish to create a new data structure to represent switches between routing resources,
+     * please follow the flyweight pattern by linking your switch ids to edges only!
+     */
+
+    inline const t_rr_switch_inf& rr_switch_inf(RRSwitchId switch_id) const {
```

Comment:
Here we will have like
inline void add_rr_switch_inf (t_rr_switch_inf);
Am I right?
So we are not going to use strong Id here?

Reply:
We will use ``StrongId`` in query APIs. For the API  ``add_rr_switch_inf()``, we can return the ``StrongId``:

```
RRSwitchId add_rr_switch_inf(t_segment_inf);
```

Reply:
I am bit confusing how we are going to create API of add_switch_inf and rr_segments.
Can you Please help me bit by showing example of anyone from both.?

Reply:
You can refer to the examples here:

https://github.com/lnis-uofu/OpenFPGA/blob/2565b64d539b04759d7b2dce51a7d061ea7a9990/vpr/src/device/rr_graph_obj.h#L639

https://github.com/lnis-uofu/OpenFPGA/blob/2565b64d539b04759d7b2dce51a7d061ea7a9990/vpr/src/device/rr_graph_obj.cpp#L904-L912